programs/mmm/src/instructions/mip1/sol_mip1_fulfill_buy.rs

```diff
@@ -205,6 +218,28 @@ pub fn handler<'info>(
     )
     .map_err(|_| MMMErrorCode::NumericOverflow)?;
 
+    // check creator_accounts and verify the remaining accounts
+    let creator_accounts = if pool.using_shared_escrow() {
+        check_remaining_accounts_for_m2(remaining_accounts, &pool.owner.key())?;
+
+        let amount: u64 = (total_price as i64 + maker_fee) as u64;
+        withdraw_m2(
+            pool,
+            ctx.bumps.pool,
+            buyside_sol_escrow_account,
+            index_ra!(remaining_accounts, 1),
+            system_program,
+            index_ra!(remaining_accounts, 0),
+            pool.owner,
+            amount,
+        )?;
+        pool.shared_escrow_cap = pool.shared_escrow_cap.checked_sub(amount).unwrap();
```

Comment:
If we want to have `shared_escrow_cap` represent the amount in SOL that we want to restrict the pool to, we need to subtract `total_price` here

Reply:
amount is total_price + fee right? in line 225

Reply:
ah ok, was confused with asset_amount in args - my bad!